usr/bs_hyc.c

```diff
@@ -144,6 +148,52 @@ static int bs_hyc_unmap(struct bs_hyc_info* infop, struct scsi_lu* lup,
 	return HycScheduleTruncate(infop->vmdk_handle, cmdp, bufp, length);
 }
 
+static int bs_hyc_sync(struct bs_hyc_info* infop, struct scsi_lu* lup,
+		struct scsi_cmd* cmdp)
+{
+	uint64_t lba;      // start lba
+	uint32_t num_blks; // end lba
+	int      result = SAM_STAT_GOOD;
+	uint8_t  key;
+	uint16_t asc;
+	uint16_t blk_shift = lup->blk_shift;
+
+	/* IMMED bit is set but it's not supported by device server */
+	if (cmdp->scb[1] & 0x2) {
+		asc = ASC_INVALID_FIELD_IN_CDB;
+		goto sense;
+	}
+
+	lba = scsi_rw_lba(cmdp->scb);
+	num_blks = scsi_rw_count(cmdp->scb);
```

Comment:
I think this should be part of scsi_cmd_offset() and scsi_cmd_length() API. We should use single API to get offset and length, the API should switch and return correct value.

Reply:
Done